pydra/engine/helpers.py

```diff
@@ -156,7 +156,7 @@ def copyfile_workflow(wf_path, result):
     for field in attr_fields(result.output):
         value = getattr(result.output, field.name)
         new_value = _copyfile_single_value(wf_path=wf_path, value=value)
-        if new_value != value:
+        if not (new_value is value or new_value == value):
```

Comment:
I am afraid that this won't work for numpy arrays `new_value` and `value` that are different from each other.

It will result in a boolean array (same size results in elementwise comparison) or raises an exception in the future (elementwise comparison is deprecated for arrays of different sizes).

Reply:
this is why I added `new_value is value` part for numpy arrays. 
Do you have an example that doesn't work with it? Could you please post it. I believe the one you send in #270 should work (added very similar test)

Reply:
The `is` only tests if `new_value` and `value` pointing to the same object. Thus, for the case of identity your code will work. But let's assume there are different objects:

```python
value = np.array([1,2,3])
new_value = np.array([4,5,6])

if not (new_value is value or new_value == value):
    print("it fails before reaching this line")
```

raises the following exception:

`ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`

Reply:
@daniel-ge - sorry, but I've missed the main problem that the numpy arrays should not be here at all. I will review the code more carefully and post a new PR